### PR TITLE
Add Node.js on Docker guide

### PR DESCRIPTION
**Pull request by Starefossen**
A simple guide to show the user how to get a Node.js application running in a Docker container using the official Docker Image for Node.js. It assumes a working installation of Docker and rootamentary understanding of how a Node.js application is structured.
- [x] short intro to Docker?
- [x] create a simple Node.js web application
- [x] build a Docker Image for the application
- [x] run the Docker Image as a container
- [x] where to get help?

This is based on the Node.js Web App guide at [doc.docker.com](http://docs.docker.com/engine/examples/nodejs_web_app/).

// cc @nodejs/docker

### CODE REVIEW

**Comment by fhemberger** [comment]
Nice, looking forward to it!

**Comment by ghost** [comment]
added some minor nitpicks, but otherwise it looks nice

**Comment by rnsloan** [comment]
Worth mentioning towards the bottom that if you want to enter the container to take a look around you can with the `exec` command `docker exec -it <container id> bash` ?

**Comment by retrohacker** [comment]
This is super slick @Starefossen :-)

I have a similar writeup here for both Docker and docker-compose: https://nodesource.com/blog/dockerizing-your-nodejs-applications
https://nodesource.com/blog/dockerizing-your-nodejs-infrastructure

**Comment by retrohacker** [comment]
I have been thinking about turning this into a blog post for a while. Perhaps it would make a better wiki? I'd like to see something focused on using Docker and docker-compose for local dev. If you think this is a good fit, I'll draft something up :smile:

https://www.youtube.com/watch?v=SUNxn0omNi4

**Comment by Starefossen** [comment]
Thanks for good feedback! I have updated the guide with everything suggested so far. There are two todos lest; _intro to docker_ and _where to get help_.

@wblankenship I have not had Node installed on my computer for the past year, I have been running everything through Docker Compose! I feel Compose may be a bit much/advanced for this guide, I would definitely like to see that go into a wiki or a separate guide!

File: locale/en/docs/guides/nodejs-docker-webapp.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    Content-Length: 12
+    Date: Sun, 02 Jun 2013 03:53:22 GMT
+    Connection: keep-alive
+
+    Hello world
+
+We hope this tutorial helped you get up and running a simple Node.js application
+on Docker.
+
+> **@TODO:** where to get help?
```
Comment (rnsloan) [inline locale/en/docs/guides/nodejs-docker-webapp.md]: Suggestions:
- official documentation: https://docs.docker.com/
- docker tag on stack overflow: http://stackoverflow.com/questions/tagged/docker
- docker subreddit: https://reddit.com/r/docker

Comment (Starefossen) [inline locale/en/docs/guides/nodejs-docker-webapp.md]: Thanks for the suggestions @rnsloan. I have pushed a new version.

**Comment by Starefossen** [comment]
I am pretty happy with the current version of the Node.js on Docker guide now. Please, nitpick and correct my spelling mistakes if you find anything, and post your suggestions on how to improve this guide.

// cc @nodejs/docker

File: locale/en/docs/guides/nodejs-docker-webapp.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+CMD [ "npm", "start" ]
+```
+
+## Building your image
+
+Go to the directory that has your `Dockerfile` and run the following command to
+build the Docker image. The `-t` flag lets you tag your image so it's easier to
+find later using the `docker images` command:
+
+    $ docker build -t <your username>/node-web-app .
```
Comment (stevemao) [inline locale/en/docs/guides/nodejs-docker-webapp.md]: use ```bash instead of indentation?

File: locale/en/docs/guides/nodejs-docker-webapp.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
+---
+title: Dockerizing a Node.js web app
+layout: docs.hbs
+---
+
+# Dockerizing a Node.js web app
+
+The goal of this example is to show you how to get a Node.js application into a
+Docker container. It assumes you have a working Docker installation and a basic
```
Comment (rnsloan) [inline locale/en/docs/guides/nodejs-docker-webapp.md]: > Docker installation

Could change that text to a link to here: https://docs.docker.com/engine/installation/

Comment (Starefossen) [inline locale/en/docs/guides/nodejs-docker-webapp.md]: :ballot_box_with_check: Fixed

File: locale/en/docs/guides/nodejs-docker-webapp.md (hunk 2 above)
Comment (Starefossen) [inline locale/en/docs/guides/nodejs-docker-webapp.md]: :ballot_box_with_check: Fixed

**Comment by Starefossen** [comment]
Well, I guess all the suggestions and feedback have been resolved. It should be good to merge now :)

**Comment by phillipj** [comment]
It would be a shame if we didn't link to this anywhere. Guess adding a link in our guides section `/docs/guides/` would be most appropriate.

**Comment by Starefossen** [comment]
Isn't that done automatic: https://github.com/nodejs/nodejs.org/blob/26e4fe006fd5d082dee39470732b08adca406a45/build.js#L110

**Comment by phillipj** [comment]
> Isn't that done automatic

Sorry, forgot guides was made as a collection.

**Comment by phillipj** [comment]
Thanks! :+1:

**Pull request merged**